source/speech/speech.py

```diff
@@ -950,6 +951,18 @@ def speak(  # noqa: C901
 		return
 	if _speechState.isPaused:
 		cancelSpeech()
+	if _speechState.speechMode == SpeechMode.onDemand:
+		
+		import inputCore
+		from scriptHandler import getCurrentScript
+		from .sayAll import SayAllHandler
+		script = getCurrentScript()
+		if not (

```

Comment:
is it worth considering speaking high priority speech (e.g. speechPriority.NOW) and speech from `ui.message` in this mode?

Reply:
@seanbudd Are you suggesting a checkbox in the UI to include high priority messages, or are you suggesting including them always?
In the general case, I think the answer to your question is "no". One of the purposes for this mode, is to prevent the screen reader from blurting things out at inconvenient times. Imagine you're doing some recording through a microphone, and suddenly an item that NVDA considers high priority pops up? If I'm using this feature, I've already decided that nothing the computer could possibly say, is more important than the reason I want it being quiet. And yet, I still want the ability to quickly use screen review or similar commands.

Yes, it means that I might miss a low battery notification or such, but if that's important to me there's probably a Windows sound I can rely upon to give me the idea that something critical is happening, without speech suddenly coming out of nowhere.

Although for the magnification user, perhaps the needs are different? (@lukaszgo1)

Reply:
> is it worth considering speaking high priority speech (e.g. speechPriority.NOW) and speech from ui.message in this mode?

Not at all. This mode means that NVDA is not speaking unless when calling commands that only report information, i.e.
perform no other action.
* High priority speech can occur at any time while the user press no command at all, e.g. alert event. This would defeat the goal of this PR. E.g. I am in a meeting and want to hear no speech feedback unless when I ask one explicitely.
* `ui.message` is used in many more script that do not belong to "on demand" mode, e.g. toggle a setting (e.g. audio ducking), perform mouse action (e.g. left click), etc.

Reply:
I agree with what been said by both @CyrilleB79  and @XLTechie  here. Also to be extra clear I am not a magnification user, so my opinion's in that regard are mostly brain dumps on how someone in that situation may want to use NVDA, and should not be considered anything conclusive.